### PR TITLE
Devcontainer updates

### PR DESCRIPTION
**Pull request by noelbundick-msft**
# Description

This PR updates the devcontainer so that it works out of the box with VS Code Dev Containers / GitHub Codespaces

[Current commit](https://github.com/cse-labs/ratify/commit/942a1f646f8e3f638514c16238a40dfd90a786ee) failed with

```
2022-10-26 17:17:29.865Z: $ /bin/sh -c sudo ./.devcontainer/post-create.sh
2022-10-26 17:17:30.146Z: sudo: ./.devcontainer/post-create.sh: command not found
2022-10-26 17:17:30.146Z: Stop (271 ms): Run in container: /bin/sh -c sudo ./.devcontainer/post-create.sh
2022-10-26 17:17:30.146Z: postCreateCommand failed with exit code 1. Skipping any further user-provided commands.
```

## Details

- Moves software installs, which require sudo, from `post-create.sh` -> `Dockerfile`
- Uses [ghcr.io/devcontainers/features/kubectl-helm-minikube](https://github.com/devcontainers/features/tree/main/src/kubectl-helm-minikube) feature rather than manually installing Kubectl/Helm
  - This is known working/validated code maintained by the VS Code team
  - The feature pulls in bash completions + VS Code extensions
  - Minikube binary is installed but not run. No conflict w/ KinD
- Removes apt installs for packages already installed in `mcr.microsoft.com/vscode/devcontainers/go:0-1.19-bullseye`
- Runs `go mod tidy` + `make` on first run. All dependencies are always & automatically pulled, present, & ready to go
- Marked `post-create.sh` as executable

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Rebuilt a new devcontainer from the updated branch, and it succeeded

### CODE REVIEW

**Review by Joll59 — APPROVED** [review]
lgtm

File: .devcontainer/post-create.sh [hunk 1]
```diff
@@ -1,47 +1,8 @@
...
-
-curl -Lo bats.tar.gz https://github.com/bats-core/bats-core/archive/v${BATS_VERSION}.tar.gz 
-
-mkdir -p bats-install/
-tar -zxf bats.tar.gz -C bats-install/
-bash bats-install/bats-core-${BATS_VERSION}/install.sh /usr/local
-
-rm -rf bats.tar.gz bats-install/
+go mod tidy
+make
```
Comment (Joll59) [inline .devcontainer/post-create.sh:8]: the make command breaks however everything up to that point works as intended.  I dont think the make command breaking is part of this PR though.

**Review by Joll59 — APPROVED** [review]

File: .devcontainer/post-create.sh (hunk 1 above)
Comment (noelbundick-msft) [reply .devcontainer/post-create.sh:8]: Should probably remove `make` until we know it's in a good state

I seem to have lost permissions on the repo - even to my own branch/PR

Would you be able to help me remove that / hit merge / etc?

**Review by Joll59 — APPROVED** [review]

**Review by cmaclaughlin — APPROVED** [review]
I get `command not found` with the updated path of `.devcontainer/post-create.sh` but since all that is executed in that script is `go mod tidy` I'm not worried about it.

**Pull request merged**